conda/cli/common.py

```diff
@@ -43,18 +44,22 @@ def confirm(message="Proceed", choices=('yes', 'no'), default='yes'):
             return choices[user_choice]
 
 
-def confirm_yn(message="Proceed", default='yes'):
-    if context.dry_run:
+def confirm_yn(message="Proceed", default='yes', dry_run=NULL):
+    dry_run = context.dry_run if dry_run is NULL else dry_run
+    if dry_run:
         from ..exceptions import DryRunExit
         raise DryRunExit()
     if context.always_yes:
         return True
     try:
-        choice = confirm(message=message, choices=('yes', 'no'),
+        choice = timeout(30, confirm, message=message, choices=('yes', 'no'),
```

Comment:
The timeout should be configurable and, I guess, infinite on default for now.
It could be helpful to prevent complete stalling in unattended execution. But other times it could be annoying:
Even though `conda >=4.4` has a considerably faster solving phase, there still might be situations where a user isn't actively waiting and the "grab-a-coffee-time" might exceed 30 seconds :wink:.

Reply:
I'm starting to become sensitive to configuration parameter fatigue.  This is trivial enough that I'd rather not even use the timeout wrapper here then if we think it's something that needs to be configurable.

Reply:
> I'm starting to become sensitive to configuration parameter fatigue.

Understandable. At some point this becomes nightmare fuel :fearful: